src/views/Dashboard/Customer/Home/RequestTest/PaymentReceiptDialog.vue

```diff
@@ -147,10 +147,23 @@ export default {
       metamaskWalletAddress: (state) => state.metamask.metamaskWalletAddress,
       metamaskWalletBalance: (state) => state.metamask.metamaskWalletBalance,
       mnemonicData: state => state.substrate.mnemonicData,
-      walletBalance: (state) => state.substrate.walletBalance
+      walletBalance: (state) => state.substrate.walletBalance,
+      web3: (state) => state.metamask.web3
     })
   },
 
+  watch: {
+    lastEventData() {
+      if (this.lastEventData) {
+        if (this.lastEventData.method === "OrderPaid") {
+          this.isLoading = false
+          this.password = ""
+          this.$router.push({ name: "customer-request-test-success"}) 
+        }
+      }      
+    }
```

Comment:
From now, let us use value inside watch events like
lastEventData(val)
so we don't need to call `this.lastEventData` over and over again